### DISCUSSION TITLE
How to use su2 in my code?

### DISCUSSION THREAD

**Discussion by zhang-qiang-github** [su2code/SU2#1864] [General]
I have successfully build `su2` from the source code, how can I use it  in my code? Could it be possible to provide a `helloworld` example? For example:

```
#include "SU2_CFD.h"

int main()
{
    SU2_CFD cfd;
    std::cout<<"helloworld"<<std::endl;
}
```

How can I include `su2` in my project rather than use it in command `SU2_CFD.exe inv_NACA0012.cfg`.

How can I build such a `helloworld` project?

**Comment by pcarruscag** [comment]
Have a look at the main function defined in SU2_CFD.cpp

**Reply by zhang-qiang-github** [reply]
Thanks for kindly reply. Could you please recommand a `IDE` to build a `su2` project? 

Now, I am using `vscode` to build a `su2` project. My steps is:

```
cd SU2/SU2_CFD/src
code .
```

![image](https://user-images.githubusercontent.com/29271051/210130783-76603e3d-7413-4c2f-a076-6c97e06d9ba9.png)

A bug is reported: `CLI11.hpp: No such file or directory`

How can I run it in `vscode`? Or should I use other `IDE`?

**Reply by pcarruscag** [reply]
I edit the code with vscode but build with meson + ninja

**Reply by zhang-qiang-github** [reply]
> I edit the code with vscode but build with meson + ninja

Is it possible to debug su2 code in the way stepping into source code? I hope I can step into source code so that I can learn the details with su2.

**Reply by pcarruscag** [reply]
Yes, you need to compile SU2 with debug information
./meson.py build --buildtype=debugoptimized ...
And then consult the vscode documentation on debugging https://code.visualstudio.com/docs/editor/debugging#_launch-configurations

**Reply by zhang-qiang-github** [reply]
Thanks, I have successfully step into `SU2_CFD.cpp`.
